sfa_dash/static/js/table-search.js

```diff
@@ -1,29 +1,107 @@
-//JQuery Table searching function 
-// Source:  https://codepen.io/adobewordpress/pen/gbewLV
+/* Jquery for filtering and searching tables.
+ * Defines filter functions that return a jquery object list of items that need to be hidden
+ * from the table. The functions should be added to the `toHideFns` array to be called by
+ * callback functions. At the end of the file, your widgets should have their callback set to
+ * the `applyTableFilters` function
+ *
+ * The containsi filter was borrowed from the below codepen:
+ * https://codepen.io/adobewordpress/pen/gbewLV
+ */
 $(document).ready(function() {
-  $(".search").keyup(function () {
-    var searchTerm = $(".search").val();
-    var listItem = $('.results tbody').children('tr');
-    var searchSplit = searchTerm.replace(/ /g, "'):containsi('")
-    
-  $.extend($.expr[':'], {'containsi': function(elem, i, match, array){
-        return (elem.textContent || elem.innerText || '').toLowerCase().indexOf((match[3] || "").toLowerCase()) >= 0;
-    }
-  });
-    
-  $(".results tbody tr").not(":containsi('" + searchSplit + "')").each(function(e){
-    $(this).attr('visible','false');
-  });
-
-  $(".results tbody tr:containsi('" + searchSplit + "')").each(function(e){
-    $(this).attr('visible','true');
-  });
-
-  var jobCount = $('.results tbody tr[visible="true"]').length;
-    $('.counter').text(jobCount + ' item');
+    /* Add containsi filter to JQuery selectors that selects for any word in a phrase */
+    $.extend($.expr[':'], {'containsi': function(elem, i, match, array){
+            return (elem.textContent || elem.innerText || '').toLowerCase().indexOf((match[3] || "").toLowerCase()) >= 0;
```

Comment:
I don't fully understand this statement. might help to assign `(elem.textContent || elem.innerText || '').toLowerCase()` to its own variable. then `newvar.indexOf((match[3] || "").toLowerCase()` to another variable. hoping the variable names make it self documenting. also what is `match[3]`?